### PR TITLE
Migrating Lead School to Lead Partner - Part III 

### PR DESCRIPTION
**Pull request by adamjcolvin**
### Context
[Trello](https://trello.com/c/vgdWlWpV)

We need to rename every reference to lead_school and its variations to lead_partner. This PR deals with renaming database columns. Where possible I've tried to follow these steps: 

* Add a new “lead_partner” column to each table.
* Backfill the lead_partner column with data from the lead_school column.
* Write new data to both columns.
* Move data reads to the new column.
* Drop the old column.

### Changes proposed in this pull request
This PR deals with the first 3 steps of the process above. Once the backfilling data migration has been run, and data is being written to both columns, then we can have a separate PR to move reads to the new columns and drop the old column. 

The affected tables are these: 
* `bulk_update_recommendations_upload_rows`
* `funding_trainee_summary_rows` 
* `hesa_students` 
* `lead_school_users`
* `schools`
* `trainees`

To avoid repeating myself with keeping the various columns in sync with each other, I've put the logic in a concern called `LeadSchoolMigratable`. This allows you to define which columns to keep in sync for each table, and will keep them in sync if either one of the columns changes. 

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Review by kennyevil — APPROVED** [review]

File: app/views/guidance/_schools.html.erb
Comment (defong) [inline app/views/guidance/_schools.html.erb:1]: I'm not sure this should be deleted?


'm pretty sure it is this page
https://www.register-trainee-teachers.service.gov.uk/guidance/hesa-register-data-mapping/schools

of which this link needs correcting from 
https://www.hesa.ac.uk/collection/c23053/e/sdlead
to
https://www.hesa.ac.uk/collection/c24053/e/sdlead

**Review by defong — CHANGES_REQUESTED** [review]
Not sure why we deleting this

`app/views/guidance/_schools.html.erb`

File: app/views/guidance/_schools.html.erb
Comment (defong) [reply app/views/guidance/_schools.html.erb:1]: This is now broken https://register-pr-4551.test.teacherservices.cloud/guidance/hesa-register-data-mapping/schools

Maybe we should add a test altogether :shrug:

**Comment by adamjcolvin** [comment]
> Not sure why we deleting this
> 
> `app/views/guidance/_schools.html.erb`

@defong That's the same partial that was giving us problems after the last PR was merged. I'm not sure how the change crept in again but I've put it back now.

File: app/views/guidance/_schools.html.erb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  <tbody class="govuk-table__body"><tr class="govuk-table__row">
+    <td class="govuk-table__cell">employing_school_urn</td>
+    <td class="govuk-table__cell">Employing school</td>
+    <td class="govuk-table__cell"><a class="govuk-link" href="https://www.hesa.ac.uk/collection/c24053/e/sdemploy">SDEMPLOY</a> (School Direct employing school)</td>
+    <td class="govuk-table__cell">–</td>
+  </tr>
+  <tr class="govuk-table__row">
+    <td class="govuk-table__cell">lead_school_urn</td>
+    <td class="govuk-table__cell">Lead partner</td>
+    <td class="govuk-table__cell"><a class="govuk-link" href="https://www.hesa.ac.uk/collection/c23053/e/sdlead">SDLEAD</a> (School Direct lead school)</td>
```
[the remark this answers is not in the document]
Comment (adamjcolvin) [reply app/views/guidance/_schools.html.erb]: Done 👍

**Review by defong — APPROVED** [review]

**Pull request merged**